src/shared/utilities/timeoutUtils.ts

```diff
@@ -67,64 +83,86 @@ export class Timeout {
      * True when the Timeout has completed
      */
     public get completed(): boolean {
-        return this._completed
+        return !!this._completionReason
     }
 
     /**
      * Updates the timer to timeout in timeout length from now
      */
     public refresh() {
-        if (this._completed === true) {
+        if (this.completed) {
             return
         }
 
         // These will not align, but we don't have visibility into a NodeJS.Timeout
         // so remainingtime will be approximate. Timers are approximate anyway and are
         // not highly accurate in when they fire.
-        this.endTime = globals.clock.Date.now() + this.timeoutLength
-        this.timerTimeout = this.timerTimeout.refresh()
+        this._endTime = globals.clock.Date.now() + this._timeoutLength
+        this._timerTimeout = this._timerTimeout.refresh()
     }
 
     /**
-     * Returns a promise that times out after timeoutLength ms have passed since Timeout object initialization
-     * Use this in Promise.race() calls in order to time out awaited functions
-     * Once this timer has finished, cannot be restarted
+     * Returns a token suitable for use in-place of VS Code's {@link CancellationToken}
      */
-    public get timer(): Promise<undefined> {
-        return this.timerPromise
+    public get token(): TypedCancellationToken {
+        return this._token
     }
 
     /**
      * Returns the elapsed time from the initial Timeout object creation
      */
     public get elapsedTime(): number {
-        return (this._completed ? this.endTime : globals.clock.Date.now()) - this.startTime
+        return (this.completed ? this._endTime : globals.clock.Date.now()) - this._startTime
+    }
+
+    private final(type: CancelEvent['type'] | 'completed'): void {
```

Comment:
Maybe rename to `terminate` or something

Reply:
is `stop()` already taken or does it not make sense, or would it make sense to overload `cancel(...)`?

 Also have `complete()` (renaming it to dispose() would definitely avoid confusion here).

Reply:
`stop` makes the most sense since it's a timer.